apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -17,15 +17,36 @@ var log = logf.Log.WithName("baremetalhost-validation")
 func (host *BareMetalHost) validateHost() []error {
 	log.Info("validate create", "name", host.Name)
 	var errs []error
+	var bmcAccess bmc.AccessDetails
+
+	if host.Spec.BMC.Address != "" {
+		var err error
+		bmcAccess, err = bmc.NewAccessDetails(host.Spec.BMC.Address, host.Spec.BMC.DisableCertificateVerification)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
 
-	if err := validateRAID(host.Spec.RAID); err != nil {
+	if err := validateRAID(host.Spec.RAID, bmcAccess); err != nil {
 		errs = append(errs, err)
 	}
 
 	if err := validateBMHName(host.Name); err != nil {
 		errs = append(errs, err)
 	}
 
+	if err := validateFirmware(host.Spec.Firmware, bmcAccess); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := validateBootMACAddress(host.Spec.BootMACAddress, bmcAccess); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := validateBootMode(host.Spec.BootMode, bmcAccess); err != nil {
+		errs = append(errs, err)
+	}
+
```

Comment:
It looks like all these new validations are related to BMC, and every one repeats the nil check (so they will be conditionally active only when there's a valid bmc access). I guess it'd preferable to group all these validation funcs under a common one, in order to avoid duplicating the check for every distinct validation func.

Reply:
Good point and agree

Reply:
Done, thanks, PTAL.